contracts/Lockbox.cdc

```diff
@@ -0,0 +1,468 @@
+    their respective objects, LockedTokenManager for the shared account,
+    and TokenHolder for the unlocked account. The user calls functions
+    on TokenHolder to withdraw tokens from the shared account and to 
+    perform staking actions with the locked tokens
+
+ */
+
+import FlowToken from 0x0ae53cb6e3f42a79
+import FungibleToken from 0xee82856bf20e2aa6
+import FlowIDTableStaking from 0xFLOWIDTABLESTAKINGADDRESS
+
+import StakingProxy from 0xSTAKINGPROXYADDRESS
+
+pub contract Lockbox {
+
+    pub event SharedAccountCreated(address: Address)
+    pub event UnlockedAccountCreated(address: Address)
+
+    /// path to store the locked token manager resource 
+    /// in the shared account
+    pub let LockedTokenManagerStoragePath: Path
+
+    /// Path to store the private capability for the token
+    /// manager
+    pub let LockedTokenManagerPrivatePath: Path
+
+    /// path to store the private locked token admin link
+    /// in the shared account
+    pub let LockedTokenAdminPrivatePath: Path
+
+    /// path to store the admin collection 
+    /// in the admin account
+    pub let LockedTokenAdminCollectionStoragePath: Path
+
+    /// path to store the token holder resource
+    /// in the unlocked account
+    pub let TokenHolderStoragePath: Path
+
+    /// Public path to store the capability that allows
+    /// reading the unlock limit of an account
+    pub let UnlockLimitPublicPath: Path
+
+    /// stored in Admin account to use to increase the unlock
+    /// token limit every time a vesting release happens
+    pub resource interface TokenAdmin {
+        pub fun increaseUnlockLimit(delta: UFix64)
+    }
+
+    /// stored in the shared account to manage access to the locked token vault
+    /// and to the staking/delegating resources
+    pub resource LockedTokenManager: FungibleToken.Receiver, FungibleToken.Provider, TokenAdmin {
+    
+        /// Capability to the normal Flow Token Vault in the shared account
+        pub var vault: Capability<&FlowToken.Vault>
+
+        /// The amount of tokens that the user can withdraw. 
+        /// It is decreased when the user withdraws
+        pub var unlockLimit: UFix64
+
+        /// Optional NodeStaker resource. Will only be filled if the user
+        /// signs up to be a node operator
+        pub var nodeStaker: @FlowIDTableStaking.NodeStaker?
+
+        /// Optional NodeDelegator resource. Will only be filled if the user
+        /// signs up to be a delegator
+        pub var nodeDelegator: @FlowIDTableStaking.NodeDelegator?
+
+        init(vault: Capability<&FlowToken.Vault>) {
+            self.vault = vault
+            self.nodeStaker <- nil
+            self.nodeDelegator <- nil
+            self.unlockLimit = 0.0
+        }
+
+        destroy () {
+            destroy self.nodeStaker
+            destroy self.nodeDelegator
+        }
+
+        // FungibleToken.Receiver actions
+
+        pub fun deposit(from: @FungibleToken.Vault) {
```

Comment:
Unfortunately I don't think it's very clear that `deposit` and `withdraw` are for moving **unlocked** FLOW in and out of the lockbox.

Maybe we could make this a passthrough to a more explicitly-named function?

```
pub fun deposit(from: @FungibleToken.Vault) {
  self.depositUnlockedTokens(from: from)
}
```

Reply:
yeah that is a good idea